From 4e178ef485cf5b23030c81c61c5a436a476a59d9 Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Thu, 21 Mar 2024 10:08:12 +0100
Subject: [PATCH 1050/1141] drm/stm: ltdc: remove support of yuv pixel formats

Soc MP25 doesn't support pixel formats yuv semiplanar & planar
on layer1 only.

Change-Id: If33a2e7fa2253fb717ed3ef5291508fe89fbb86e
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/368188
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/stm/ltdc.c | 25 ++++++++++++++++---------
 1 file changed, 16 insertions(+), 9 deletions(-)

--- a/drivers/gpu/drm/stm/ltdc.c
+++ b/drivers/gpu/drm/stm/ltdc.c
@@ -1980,15 +1980,22 @@ static struct drm_plane *ltdc_plane_crea
 			       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_cp) * sizeof(*formats));
 			nb_fmt += ARRAY_SIZE(ltdc_drm_fmt_ycbcr_cp);
 		}
-		if (val & LXCR_C1R_YSPA) {
-			memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_sp,
-			       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp) * sizeof(*formats));
-			nb_fmt += ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp);
-		}
-		if (val & LXCR_C1R_YFPA) {
-			memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_fp,
-			       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_fp) * sizeof(*formats));
-			nb_fmt += ARRAY_SIZE(ltdc_drm_fmt_ycbcr_fp);
+
+		/*
+		 * Soc MP25 doesn't support pixel formats yuv semiplanar &
+		 * planar on layer1 only.
+		 */
+		if (!(of_device_is_compatible(dev->of_node, "st,stm32mp25-ltdc") && !index)) {
+			if (val & LXCR_C1R_YSPA) {
+				memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_sp,
+				       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp) * sizeof(*formats));
+				nb_fmt += ARRAY_SIZE(ltdc_drm_fmt_ycbcr_sp);
+			}
+			if (val & LXCR_C1R_YFPA) {
+				memcpy(&formats[nb_fmt], ltdc_drm_fmt_ycbcr_fp,
+				       ARRAY_SIZE(ltdc_drm_fmt_ycbcr_fp) * sizeof(*formats));
+				nb_fmt += ARRAY_SIZE(ltdc_drm_fmt_ycbcr_fp);
+			}
 		}
 	}
 
